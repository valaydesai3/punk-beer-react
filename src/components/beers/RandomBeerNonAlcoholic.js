import React, { useContext } from 'react';
import PunkBeerContext from '../../context/punkBeerContext';

const RandomBeerNonAlcoholic = () => {
  const punkBeerContext = useContext(PunkBeerContext);
  return (
    <div data-testid="nonAlcoBeerBtn">
      <button
        className="btn"
        onClick={() => punkBeerContext.getRandomBeer(true)}
      >
        Rnadom non-alcoholic beer
      </button>
    </div>
  );
};

export default RandomBeerNonAlcoholic;
