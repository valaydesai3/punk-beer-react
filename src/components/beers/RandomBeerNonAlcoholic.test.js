// __tests__/fetch.test.js
import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import PunkBeerContext from '../../context/punkBeerContext';
import RandomBeerNonAlcoholic from './RandomBeerNonAlcoholic';

afterEach(cleanup);

test('renders non-alocholic beer button', async () => {
  let getRandomBeer = jest.fn();

  const { getByText, getByTestId } = render(
    <PunkBeerContext.Provider
      value={{
        getRandomBeer
      }}
    >
      <RandomBeerNonAlcoholic />
    </PunkBeerContext.Provider>
  );
  expect(getByTestId('nonAlcoBeerBtn')).toHaveTextContent(
    'Rnadom non-alcoholic beer'
  );

  fireEvent.click(getByText('Rnadom non-alcoholic beer'));
  expect(getRandomBeer).toHaveBeenCalled();
  expect(getByTestId('nonAlcoBeerBtn')).toMatchSnapshot();
});
