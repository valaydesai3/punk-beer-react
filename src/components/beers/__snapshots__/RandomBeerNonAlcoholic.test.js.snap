// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders non-alocholic beer button 1`] = `
<div
  data-testid="nonAlcoBeerBtn"
>
  <button
    class="btn"
  >
    Rnadom non-alcoholic beer
  </button>
</div>
`;
